### PR TITLE
Added UIUserInterfaceIdiom type definitions

### PR DESCRIPTION
**Pull request by dlevi309**
Let me know if something should be changed

### CODE REVIEW

File: UIKit/UIUserInterfaceIdiom.h [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#import <UIKit/UIKit.h>
+#import <Foundation/Foundation.h>
```
Comment (leptos-null) [inline UIKit/UIUserInterfaceIdiom.h:2]: nit: not sure if we need both UIKit and Foundation

**Comment by leptos-null** [comment]
Are these not [public](https://developer.apple.com/documentation/uikit/uiuserinterfaceidiom?language=objc)?

**Comment by dlevi309** [comment]
OMG, you’re right. I’d gotten so used to using it that I had’t even realized. After checking, the Apple Watch declaration seems to be missing though from the public headers, but you’re correct.

I can close this PR if you think that the Apple Watch def isn’t worth adding?

**Comment by leptos-null** [comment]
I think the Apple Watch idiom should be added. I'm not sure how to, though. re-declaring the enum is an error.

```objc
enum { UIUserInterfaceIdiomWatch = 4 };
```
seems to work, but then it's kind of untyped.

**Comment by dlevi309** [comment]
how about an additional constant to UIDevice like;

```objc
static const UIUserInterfaceIdiom UIUserInterfaceIdiomWatch = (UIUserInterfaceIdiom)4;
```

**Comment by dlevi309** [comment]
got it, thanks PS

**Review by PoomSmart — APPROVED** [review]

**Pull request merged**